packages/makecode-core/src/commands.ts

```diff
@@ -919,27 +919,62 @@ export async function shareCommand(opts: ProjectOptions) {
 }
 
 export async function getSimHTML(opts: ProjectOptions) {
-    applyGlobalOptions(opts);
-
-    const project = await resolveProject(opts);
-    const cache = await project.getCacheAsync();
-
-    const key = project.editor.website + "-sim.html"
-    return getExpandedPageFromCache(key, cache, project.editor.website);
+    return getExpandedHTML(opts, "cachedSimulatorKey", "sim");
 }
 
 export async function getAssetEditorHTML(opts: ProjectOptions) {
+    return getExpandedHTML(opts, "cachedAssetEditorKey", "asseteditor");
+}
+
+async function getExpandedHTML(opts: ProjectOptions, downloadInfoKey: "cachedSimulatorKey" | "cachedAssetEditorKey", pageName: "sim" | "asseteditor") {
```

Comment:
Will there be a scenario where these union types will want to be expanded in the future? If so, can we put these in a separate type so if we do expand them, it doesn't make the function definition more busy?

Reply:
i don't think so, but we can cross that bridge when we come to it